dotcom-rendering/src/lib/article-count.ts

```diff
@@ -0,0 +1,32 @@
+import { storage } from '@guardian/libs';
+import {
+	getWeeklyArticleHistory,
+	incrementWeeklyArticleCount,
+} from '@guardian/support-dotcom-components';
+import { hasOptedOutOfArticleCount } from '../web/lib/contributions';
+import { incrementDailyArticleCount } from '../web/lib/dailyArticleCount';
+
+// Log an article view using the Slot Machine client lib
+// We should monitor this function call to ensure it only happens within an
+// article pages when other pages are supported by DCR.
+export const getArticleCount = async (pageId: string, keywordIds: string) => {
+	const hasOptedOut = await hasOptedOutOfArticleCount();
+
+	// hasOptedOut could potentially be moved into the scope of the below if condition
```

Comment:
Does anybody know enough about how async works in JS that could confirm this?

Reply:
I think you are correct.
The `window.guardian.articleCount` check must happen after the await.
Otherwise if two invocations await at the same time then both would go onto increment and update `window.guardian.articleCount`

Reply:
Worth updating this comment to be a statement rather than a question?